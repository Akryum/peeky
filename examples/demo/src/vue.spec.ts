/* @peeky {
  runtimeEnv: 'dom'
} */

// Include template compiler
import { createApp } from 'vue/dist/vue.esm-bundler'

describe('vue', () => {
  test('create vue app', () => {
    const app = createApp({
      data () {
        return {
          msg: 'hello',
        }
      },
      template: '<div>{{ msg }}</div>',
    })
    console.log(app)
    expect(typeof app.version).toBe('string')
    const el = document.createElement('div')
    document.body.appendChild(el)
    app.mount(el)
    expect(document.body.innerHTML).toMatchSnapshot()
    expect('Hello').toMatchSnapshot('demo')
    expect('Meow').toMatchSnapshot('demo')
    // expect(el.innerHTML).toBe('<div>hello</div>')
  })
})
