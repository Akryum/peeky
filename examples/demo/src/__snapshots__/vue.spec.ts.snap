// Peeky snapshots v1.0

exports[`vue create vue app 1`] = `
<div data-v-app>
  <div>
    hello
  </div>
</div>
`;

exports[`vue create vue app demo 1`] = `
"Hello"
`;

exports[`vue create vue app demo 2`] = `
"Meow"
`;